File: src/main/java/com/enderryno/nuclearcraft/enums/StructureBehaviour.java
package com.enderryno.nuclearcraft.enums;

import com.enderryno.nuclearcraft.classes.structures.Warhead;
import com.enderryno.nuclearcraft.interfaces.PluginStructure;
import org.apache.maven.model.Plugin;

public enum StructureBehaviour {
    WARHEAD("WARHEAD"),
    ;

    final String name;
    StructureBehaviour(String name) {
        this.name = name;
    }

    public static StructureBehaviour fromString(String text) {
        for (StructureBehaviour b : StructureBehaviour.values()) {
            if (b.name.equalsIgnoreCase(text)) {
                return b;
            }
        }
        return null;
    }

    public Class<? extends PluginStructure> getStructureClass() {
        switch (this) {
            case WARHEAD:
                return Warhead.class;
            default:
                return null;
        }
    }

    public String getName() {
        return name;
    }
}

File: src/main/java/com/enderryno/nuclearcraft/enums/StructureBehaviour.kt
package com.enderryno.nuclearcraft.enums

import com.enderryno.nuclearcraft.classes.structures.Bomb
import com.enderryno.nuclearcraft.interfaces.PluginStructure


enum class StructureBehaviour(name: String) {
    BOMB("BOMB");

    val structureClass: Class<out PluginStructure?>?
        get() = when (this) {
            BOMB -> Bomb::class.java
            else -> null
        }

    companion object {
        fun fromString(text: String?): StructureBehaviour? {
            for (b in values()) {
                if (b.name.equals(text, ignoreCase = true)) {
                    return b
                }
            }
            return null
        }
    }
}
